### PR TITLE
apollo: support proxy env vars

### LINKED ISSUE #523: schema publish error: read ECONNRESET
hi there, i am using the new apollo-cli to upload schema  but aways get `ECCONNRESET` error: 

image

### LINKED ISSUE #1793: Due to the Chinese network environment, blocked by the firewall, the execution of the npm run codegen command failed
I have set up a network ladder and set up a proxy for the terminal, but it still does n’t work. Excuse me, are there any other options?
屏幕快照 2020-02-07 上午9 53 12

### LINKED ISSUE #1967: Support GLOBAL_AGENT_HTTP_PROXY for apollo CLI
The apollo module should be updated to support GLOBAL_AGENT_HTTP_PROXY like apollo-server. Because it doesn't I currently have to whitelist an egress to engine-graphql.apollographql.com from a secure cluster in order to upload my schema to graph manager during CI.

### PR DESCRIPTION
**Pull request by glasser**
Fixes #523. Fixes #1967. Fixes #1793.

### CODE REVIEW

**Comment by glasser** [comment]
My testing plan is to make RCs and ask people on the issues above to test them. I have verified that basic commands appear to work without the environment variables set and that setting the env vars to nonexistent URLs leads to expected errors.

**Comment by glasser** [comment]
For what it's worth I would prefer a more explicit approach where we make an agent object and pass it as fetch options in the few places where we do HTTP. But surprisingly there does not appear to be a major npm library that exports "an Agent that parses the standard environment variables without installing the Agent onto http.globalAgent". Ah well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Upcoming
 
 - `apollo`
-  - <First `apollo` related entry goes here>
+  - Support the standard `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables for using an HTTP proxy.
```
Comment (abernix) [inline CHANGELOG.md:6]:
```suggestion
  - Support the standard `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables for using an HTTP proxy. [#2181](https://github.com/apollographql/apollo-tooling/pull/2181/)
```

**Review by abernix — APPROVED** [review]
Thanks!

**Pull request merged**